torch/_meta_registrations.py

```diff
@@ -20,6 +20,7 @@
     ELEMENTWISE_TYPE_PROMOTION_KIND,
     IntLike,
     make_contiguous_strides_for,
+    NumberType,
```

Comment:
All changes in this file are to help make sense of overloads. They match macros on cpu side. FOREACH_BINARY_OP_TENSOR etc. Might switch to the standard way of defining metas once the tests pass.